app/controllers/solutions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SolutionsController < ApplicationController
+  def create
+    @question = Question.find(params[:question_id])
+    @solution = @question.solutions.build(solution_params)
+    @solution.question_id = @question.id
+    if @solution.save
+      redirect_to @question, notice: "回答を投稿しました。"
+    else
+      @solutions = Solution.all
```

Comment:
これでは，レンダリング時に他の質問の回答も表示されてしまいます。
showアクションで定義した`@solutions`を与える必要があります。

```suggestion
      @solutions = @question.solutions
```

Reply:
`@solutions = @question.solutions` へ修正しました。